elastic/kibana
# [Index Management] Fix index template simulate request

Fixes https://github.com/elastic/kibana/issues/165889

## Summary

This PR fixes the simulate request for index templates. Previously, it would fail when we simulate an index template which references a component template that is not created yet, because the `ignore_missing_component_templates` option wasn't configured. This PR adds this property to the template format so that it is sent as part of the simulate request.

**How to test:**
1. Start Es with `yarn es snapshot` and Kibana with `yarn start`.
2. Go to Dev Tools.
3. Create an index template that references a component template:
```
PUT _index_template/test-template
{
  "index_patterns": ["test-*"],
  "composed_of": ["some_component_template"],
  "ignore_missing_component_templates": ["some_component_template"]
}
```
4. Go to Stack Management -> Index Management -> Index Templates.
5. Click on the created index template to open the details flyout.
6. Click on the "Preview" tab and verify that the simulate request was successful.
